test/integration/http_protocol_integration.cc

```diff
@@ -13,23 +13,27 @@ std::vector<HttpProtocolTestParams> HttpProtocolIntegrationTest::getProtocolTest
       for (auto upstream_protocol : upstream_protocols) {
 #ifdef ENVOY_ENABLE_QUIC
         ret.push_back(
-            HttpProtocolTestParams{ip_version, downstream_protocol, upstream_protocol, false});
+            HttpProtocolTestParams{ip_version, downstream_protocol, upstream_protocol, kBareHttp2});
         if (downstream_protocol == Http::CodecType::HTTP2 ||
             upstream_protocol == Http::CodecType::HTTP2) {
+          ret.push_back(HttpProtocolTestParams{ip_version, downstream_protocol, upstream_protocol,
+                                               kWrappedHttp2});
           ret.push_back(
-              HttpProtocolTestParams{ip_version, downstream_protocol, upstream_protocol, true});
+              HttpProtocolTestParams{ip_version, downstream_protocol, upstream_protocol, kOgHttp2});
         }
 #else
         if (downstream_protocol == Http::CodecType::HTTP3 ||
             upstream_protocol == Http::CodecType::HTTP3) {
           ENVOY_LOG_MISC(warn, "Skipping HTTP/3 as support is compiled out");
         } else {
-          ret.push_back(
-              HttpProtocolTestParams{ip_version, downstream_protocol, upstream_protocol, false});
+          ret.push_back(HttpProtocolTestParams{ip_version, downstream_protocol, upstream_protocol,
+                                               kBareHttp2});
```

Comment:
Here and above, it somewhat strange why `kBareHttp2` is used, when it might be H1.
I guess it makes sense for the test, though.

Reply:
Yeah, HttpProtocolTestParams needs a fourth value. If you have other suggestions, I'm open to them. Maybe {} would work?